probe_scraper/runner.py

```diff
@@ -257,6 +257,38 @@ def load_moz_central_probes(
     write_moz_central_probe_data(probes_by_channel_with_dates, revision_dates, out_dir)
 
 
+def add_source_url(results, url, filepath):
+    for result in results:
+        line_number = results[result]["defined_in"]["line"]
+        results[result]["source_url"] = f"{url}/blob/{filepath}#L{line_number}"
+        # the 'defined_in' structure is no longer needed
+        del results[result]["defined_in"]
+    return results
+
+
+def get_file_path_with_hash(path, commit_hash):
+    """
+    Get the relative file path with commit hash included
+    from disk path (to be used with GitHub repo link to
+    create a source url)
+
+    The structure of disk path is "base_path / _hash / rel_path"
+    We want to extract _hash and rel_path from this string.
+
+    Example input:
+    '/var/folders/yf/q5zg5fjx2kz2lc2sbfdht4bm0000gn/T/tmpt9x11our/glean-android/
+    530a5128526fcf133d57ec3d188590c600895fa1/glean-core/metrics.yaml'
+
+    Output:
+    '530a5128526fcf133d57ec3d188590c600895fa1/glean-core/metrics.yaml'
+    """
+    split_path = "".join(path).split("/")
+    for i, word in enumerate(split_path):
+        if word == commit_hash:
+            file_path = "/".join(split_path[i:])
+    return file_path
```

Comment:
I *think* you can replace this with this one-liner:

```suggestion
    return path.split(commit_hash)[-1][1:]
```